geoips/interfaces/base.py

```diff
@@ -330,10 +330,18 @@ def __new__(cls):
 
     def __init__(self):
         """YAML plugin interface init method."""
-        # self._unvalidated_plugins = self._create_unvalidated_plugins_cache(
-        #     load_all_yaml_plugins()
-        # )
-        self._unvalidated_plugins = load_all_yaml_plugins()
+        try:
+            self._unvalidated_plugins = load_all_yaml_plugins()
+        except PluginRegistryError:
+            self._unvalidated_plugins = {}
```

Comment:
NOTE: This will just continue on if load_all_yaml_plugins raises an error.  This was to prevent create_plugin_registry from failing when attempting to import geoips.interfaces.products, before even getting to the point of writing out the registered_plugins.yaml files.

Reply:
This looks fine for now. To resolve this, we should create a PluginRegistry class that performs a delayed load for plugins. That would allow us to avoid calling load_all_yaml_plugins when we run create_plugin_registry.